test/gtest/ucs/test_sys.cc

```diff
@@ -137,3 +148,16 @@ UCS_TEST_F(test_sys, memunits_to_str) {
     test_memunits(UCS_TBYTE, "1T");
     test_memunits(UCS_TBYTE * 1024, "1024T");
 }
+
+UCS_TEST_F(test_sys, cpu_cache) {
+#if defined(__aarch64__)
+    UCS_TEST_SKIP_R("ARM");
+#endif
+
+    check_cache_type(UCS_SYS_CACHE_L1d, "L1d");
+    check_cache_type(UCS_SYS_CACHE_L1i, "L1i");
+#if defined(__x86_64__)
+    check_cache_type(UCS_SYS_CACHE_L2, "L2");
```

Comment:
on PPC systems there is incomplete cache information (missing size and level on some systems, as I can see depends from kernel build configuration). for now cache size going to be used on x86 systems only

Reply:
then I would propose to move the functionality under __x86_64__ too and return "not implemented" error for other uarchs.

Reply:
why? this functionality depends from kernel implementation and may be changed in future

Reply:
so, should not tests track that changes?

Reply:
we are tracking regressions... no ideas how to track progress :)

Reply:
i don't see a difference but ok if this is not required

Reply:
If we have to introduce #ifdef (architecture) , it does not belong to common code...

Reply:
we already using arch-specific tests, for this case it is reasonable to allow tests pass. when we have way to get cache size for other arch - I will update tests

Reply:
@hoopoepg another problem here is the upcoming freebsd and osx support. @hiroyuki-sato @kostikbel.

Reply:
FreeBSD implementation of such query for x86 would use CPUID directly anyway.  Then, it is not obvious does it make sense to query the llnux kernel on linux if CPUID is there.

Reply:
On arm you have to be in privileged mode to get access to cpuid register

Reply:
> On arm you have to be in privileged mode to get access to cpuid register

Sure, and my point is that you would use different calls on Linux vs. FreeBSD/ARMv8.  We can have common OS-agnostic code for x86.  Or we can have per-OS (and still probably per-ISA) methods.

Anyway, FreeBSD port will cope with whatever decision you make.

Reply:
Hello, @shamisp 

Thank you for telling me this PR.

OpenUCX use `/proc` and `/sys` for reading much information. I need to implement instead of this way on macOS. I think `sysctl` and `sysctlbyname(3)` usable.

```
sysctl -a | grep cachesize
hw.cachesize: 17179869184 32768 262144 4194304 0 0 0 0 0 0
hw.l1icachesize: 32768
hw.l1dcachesize: 32768
hw.l2cachesize: 262144
hw.l3cachesize: 4194304
```